src/bin/pg_autoctl/fsm.c

```diff
@@ -168,14 +172,15 @@ KeeperFSMTransition KeeperFSM[] = {
 	{ DRAINING_STATE, DEMOTED_STATE, COMMENT_DRAINING_TO_DEMOTED, &fsm_stop_postgres },
 	{ PRIMARY_STATE, DEMOTED_STATE, COMMENT_PRIMARY_TO_DEMOTED, &fsm_stop_postgres },
 	{ PRIMARY_STATE, DEMOTE_TIMEOUT_STATE, COMMENT_PRIMARY_TO_DEMOTED, &fsm_stop_postgres },
+	{ PRIMARY_STATE, MAINTENANCE_STATE, COMMENT_PRIMARY_TO_MAINTENANCE, &fsm_stop_postgres },
```

Comment:
I'm not sure how to prevent that... other than maybe going from maintenance to demote timeout. In that case pg_autoctl will ensure Postgres is shut down for as long as it runs, at least. Another way to do that might be to prepare the replication setup so that if Postgres is started, it's already setup to be a standby, already in the MAINTENANCE state.

Reply:
So I think we can still code that without extra states. As we do a clean shutdown on the primary before promoting the secondary, we don't expect to have to do a `pg_rewind` later. Given that we can share the same code for MAINTENANCE to CATCHINGUP in the case when the node used to be a primary or a secondary before entering MAINTENANCE.

If review shows that this is an ill advised position to take, then we will need to introduce a specific primary only maintenance state.